tests/src/Kernel/BaseFieldTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_contact_forms\Kernel;
+
+use Drupal\Tests\rdf_entity\Kernel\RdfKernelTestBase;
+use Drupal\contact\Entity\ContactForm;
+use Drupal\contact\Entity\Message;
+
+/**
+ * Tests contact form with new fields.
+ */
+class BaseFieldTest extends RdfKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'options',
+    'telephone',
+    'contact',
+    'contact_storage',
+    'rdf_skos',
+    'oe_contact_forms',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->installEntitySchema('contact_message');
+  }
+
+  /**
+   * Tests successful corporate forms message.
+   */
+  public function testCorporateFormBaseFields(): void {
+    $contact_form_id = 'oe_contact_form';
+    $contact_form = ContactForm::create(['id' => $contact_form_id]);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'topics', [['topic_name' => 'Topic name', 'topic_email_address' => 'topic@emailaddress.com']]);
+    $contact_form->save();
+
+    $data = [
+      'id' => 1,
+      'contact_form' => $contact_form->id(),
+      'name' => 'example',
+      'mail' => 'admin@example.com',
+      'created' => time(),
+      'ip_address' => '127.0.0.1',
+      'subject' => 'Test subject',
+      'message' => 'Test message',
+      // Add an unsupported value.
+      'oe_topic' => 1,
+    ];
+
+    $message = Message::create($data);
+    // Validate the field.
+    $violations = $message->oe_topic->validate();
+    $this->assertNotEmpty($violations);
```

Comment:
$violationsi is an object, will never be empty. Assert better the count of its violations as well as any violation messages.

Reply:
if there are no errors then false is returned

Reply:
Like I said, $violations is an object that can be interrogated. And we should be specific in our assertions and assert the count and the message of the violation.

With violations:
![Screenshot 2020-05-14 09 01 06](https://user-images.githubusercontent.com/5848933/81903123-8e7ced00-95c1-11ea-8f8b-04bcf8706c4f.png)

Without:
![Screenshot 2020-05-14 09 02 36](https://user-images.githubusercontent.com/5848933/81903203-b2d8c980-95c1-11ea-9aa6-f152c93491fe.png)